pennylane_orquestra/orquestra_device.py

```diff
@@ -0,0 +1,378 @@
+    filenames for the generated workflows are stored in the ``filenames``
+    attribute.
+
+    Computing the expectation value of the identity operator does not involve a
+    workflow submission (hence no files are created).
+
+    Args:
+        wires (int, Iterable[Number, str]]): Number of subsystems represented
+            by the device, or iterable that contains unique labels for the
+            subsystems as numbers (i.e., ``[-1, 0, 2]``) or strings (``['ancilla',
+            'q1', 'q2']``). Default 1 if not specified.
+        shots (int): number of circuit evaluations/random samples used to estimate
+            expectation values of observables
+        analytic (bool): If ``True``, the device calculates probability,
+            expectation values, and variances analytically. If ``False``, a finite
+            number of samples set by the argument ``shots`` are used to estimate
+            these quantities.
+
+    Keyword Args:
+        backend=None (str): the Orquestra backend device to use for the
+            specific Orquestra backend, if applicable
+        batch_size=10 (int): the size of each circuit batch when using the
+            ``~.batch_execute`` method to send multiple workflows
+        keep_files=False (bool): Whether or not the workflow files
+            generated during the circuit execution should be kept or deleted.
+        resources=None (dict): An option for Orquestra, specifies the resources
+            provisioned for the clusters running each workflow step.
+        timeout=300 (int): The time until a job should timeout after getting no
+            response from Orquestra (in seconds).
+    """
+
+    name = "Orquestra base device for PennyLane"
+    short_name = "orquestra.base"
+    pennylane_requires = ">=0.11.0"
+    version = __version__
+    author = "Xanadu"
+
+    operations = {
+        "BasisState",
+        "CNOT",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+        "CSWAP",
+        "CY",
+        "CZ",
+        "Hadamard",
+        "MultiRZ",
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "PhaseShift",
+        "QubitStateVector",
+        "RX",
+        "RY",
+        "RZ",
+        "Rot",
+        "S",
+        "SWAP",
+        "SX",
+        "T",
+        "Toffoli",
+    }
+
+    observables = {"PauliX", "PauliY", "PauliZ", "Identity", "Hadamard"}
+
+    def __init__(self, wires, shots=1000, analytic=True, **kwargs):
+        super().__init__(wires=wires, shots=shots, analytic=analytic)
+
+        self.backend = kwargs.get("backend", None)
+        self._batch_size = kwargs.get("batch_size", 10)
+        self._keep_files = kwargs.get("keep_files", False)
+        self._resources = kwargs.get("resources", None)
+        self._timeout = kwargs.get("timeout", 300)
+        self._latest_id = None
+        self._filenames = []
+        self._backend_specs = None
+
+    def apply(self, operations, **kwargs):
+        pass
```

Comment:
Never used, but abstract in `QubitDevice`, so it needs to be redefined.

Reply:
True, but its subclasses (`QiskitDevice`, etc.) are instantiated. Getting a `TypeError: Can't instantiate abstract class B with abstract methods apply` for
```python
import pennylane as qml

class A(qml.QubitDevice):
    author = "Author"
    name = "Name"
    operations = None
    pennylane_requires = None
    short_name = None
    version = None

class B(A):
    pass

b = B()
```
Lmk if missing something though!

Reply:
Oh, I see - none of the subclasses define `apply`. How come this is the case? I thought `apply` was required?

Reply:
The `OrquestraDevice.execute` and `OrquestraDevice.batch_execute` methods will be self-contained as they use circuit serialization on a `CircuitGraph` object as a whole, so it seemed that having an `apply` method definition might not make sense.

Reply:
Got it, something to keep in mind for when we merge QubitDevice and Device

Reply:
> The OrquestraDevice.execute and OrquestraDevice.batch_execute methods will be self-contained as they use circuit serialization on a CircuitGraph object as a whole, so it seemed that having an apply method definition might not make sense.

Perhaps worth adding the above as a comment to the `apply()` method, so that others developers understand why its defined but not used.